check/base.go

```diff
@@ -116,6 +116,10 @@ func (c *FTWCheck) GetTriggeredRules() []uint {
 	if c.CloudMode() {
 		return nil
 	}
+	// When a test is expecting to trigger an error and it effectively does, markers are not set.
```

Comment:
After https://github.com/coreruleset/go-ftw/pull/432, `GetTriggeredRules()` is now called not only when we are looking for IDs in the logs based on the test expectations but also at the end of the test to provide the list of triggered rules. When a test is expected to trigger an error, start and end markers might not populated. As a result, calling TriggeredRules() in such cases triggers a fatal error. Similar to how [we do not support looking for IDs when expecting an error](https://github.com/coreruleset/go-ftw/blob/528e92cae88cd7c084e087856214f71f687cc4dc/runner/run.go#L371-L374), I believe it is reasonable not to look for triggered rules in this case.